### ISSUE TITLE
Mac OS X Fuse / Seek support

### ISSUE THREAD

**Issue by sjezewski** [pachyderm/pachyderm#333]
Seeks are not supported on Mac OS X. Doing seeks will result in undefined behavior. As OS X implements IO, we cannot differentiate between a seek and subsequent file writes.

I'm making this issue as a 'beacon' so that:
- if people really want it, they can upvote, and
- to enable debugging weird behavior on os x

If you look at the behavior in the tests `TestSeekRead` / `TestSeekWriteGap` / `TestSeekWriteBackwards` that sort of behavior is undefined on mac.

On linux, the OS will respect the `fuse.OpenNonSeekable` flag and disallow seeks. On mac, they are allowed, and OS X doesn't really implement the direct IO interface either ... it buffers the writes in an odd way -- see below. Because they report the cumulative write, we have the manual offset detection in `filesystem.go` under the `func (h *handle) Write(...)` function. We basically cannot detect a seek vs a normal write.

For instance, in the case of `TestSeekWriteGap`, mac will issue:

Mac:

``` shell
WriteRequest: foo@0
WriteRequest: foobaz@0
```

Linux:

``` shell
WriteRequest: foo@0
WriteRequest: baz@6
```

For instance, in the case of `TestSeekWriteBackwards`, mac will issue:

Mac:

``` shell
WriteRequest: foofoofoo@0
WriteRequest: foobarfoo@0
```

Linux:

``` shell
WriteRequest: foofoofoo@0
WriteRequest: bar@3
```

**Comment by jdoliner** [comment]
I don't think there's anything to actually do here. This issue should be fixed by #529.
I'm putting it in 1.2 so we remember to check that's the case.

**Comment by sjezewski** [comment]
No one but me has run into this. Closing.
